src/test/java/com/google/devtools/build/lib/blackbox/tests/manageddirs/ManagedDirectoriesBlackBoxTest.java

```diff
@@ -392,6 +377,40 @@ public void testWorkspaceSymlinkThrowsWithManagedDirectories() throws Exception
         .isTrue();
   }
 
+  @Test
+  public void testNoCheckFiles(@TestParameter boolean allSkips, @TestParameter boolean watchFs) throws Exception {
+    generateProject();
+    buildExpectRepositoryRuleCalled(false, watchFs);
+    checkProjectFiles();
+
+    Path nodeModules = context().getWorkDir().resolve("node_modules");
+    assertThat(nodeModules.toFile().isDirectory()).isTrue();
+    PathUtils.deleteTree(nodeModules);
+    assertThat(nodeModules.toFile().isDirectory()).isFalse();
+
+    // We don't check the external file disappeared so the build is broken
+    BuilderRunner bazel = bazel(watchFs).withFlags("--noexperimental_check_external_files");
+    if (allSkips) {
+      bazel = bazel.withFlags("--noexperimental_check_output_files");
+    }
+
+    if (watchFs && OS.getCurrent() == OS.DARWIN) {
+      // On Darwin CI, --watchfs makes the build succeed
+      ProcessResult result = bazel.build("//...");
+      buildSucceeded(result);
+    } else {
+      ProcessResult result = bazel.shouldFail().build("//...");
+      assertThat(findPattern(result, "Not found package.json")).isTrue();
+    }
+
+    // it doesn't make the file on disk
+    assertThat(nodeModules.toFile().isDirectory()).isFalse();
+
+    // But in a perfect world a normal build would fix it. TODO: uncomment this
```

Comment:
I don't think we need to keep a TODO here, or commented out code. Just a comment explaining that we don't check for the missing file until something else changes.